solr/modules/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java

```diff
@@ -233,7 +235,14 @@ protected static boolean scoreSingleHit(
     boolean logHit = false;
 
     scorer.getDocInfo().setOriginalDocScore(hit.score);
+    QueryLimits queryLimits = QueryLimits.getCurrentLimits();
     hit.score = scorer.score();
+    if (queryLimits.maybeExitWithPartialResults(
+        "Learning To Rank rescoring -"
+            + " The full reranking didn't complete and got reverted."
+            + " All documents preserved their original score and ranking.")) {
+      throw new QueryLimitsExceededException("The Time Allowed has been exceeded when rescoring");
```

Comment:
It's not necessarily `timeAllowed` now ... any other limit may have been exceeded. A better wording would be "A query limit has been exceeded".

Reply:
Thanks for the correction, it's coming in the next commit.
In regards to the exception discussion, I'll talk about it in the other comment